src/tests/unit/tests/injected/analyzers/unified-rule-analyzer.test.ts

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
advice appreciated on how to / whether to share code between this and rule-analyzer. Would be nice to use composition instead but the methods in question are protected